### PR TITLE
[8.0] add stock_no_extra_move by gurneyalex

### PR DESCRIPTION
**Pull request by leemannd**
Superseeds https://github.com/OCA/stock-logistics-workflow/pull/453/files from @scigghia .

### CODE REVIEW

**Comment by leemannd** [comment]
Improvement has been proposed from OCA code sprint. Moving it back to WIP

File: stock_no_extra_move/models/stock_picking.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+class StockPicking(models.Model):
+    _inherit = 'stock.picking'
+
+    @api.model
+    def _create_extra_moves(self, picking):
+        User = self.pool['res.users']
+        if not User.has_group(
+                self.env.cr, self.env.user.id,
+                'stock_no_extra_move.group_can_increase_quantity'):
+            if picking.picking_type_id.code != 'incoming':
```
Comment (rousseldenis) [inline stock_no_extra_move/models/stock_picking.py]: @leemannd Let do an option on stock.picking.type to be as flexible as you want (and not limited to 'not incoming' ones)

Comment (leemannd) [reply stock_no_extra_move/models/stock_picking.py]: Indeed

File: stock_no_extra_move/wizard/stock_transfer_details.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+
+class StockTransferDetailsItems(models.TransientModel):
+    _inherit = 'stock.transfer_details_items'
+
+    @api.one
+    @api.constrains('quantity', 'orig_quantity')
+    def _check_quantities(self):
+        User = self.pool['res.users']
+        if self.quantity < 0:
```
Comment (rousseldenis) [inline stock_no_extra_move/wizard/stock_transfer_details.py:15]: @leemannd Please use of float_compare as quantity is decimal precision aware https://github.com/odoo/odoo/blob/9e8f70e4849b0eeaca8b5cf51372ecfa23dc561b/addons/stock/wizard/stock_transfer_details.py#L139

File: stock_no_extra_move/models/stock_picking_type.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp import fields, models
+
+
+class StockPicking(models.Model):
+    _inherit = 'stock.picking.type'
+
+    allow_process_qty = fields.Boolean(
+        string='Allow process quantities',
```
Comment (rousseldenis) [inline stock_no_extra_move/models/stock_picking_type.py:12]: @leemannd Maybe a little help here ?

File: stock_no_extra_move/wizard/stock_transfer_details.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+                ):
+                    percent_increase_quantity_allowed = self.env[
+                        'stock.config.settings'
+                    ].get_default_percent_increase_quantity_allowed()[
+                        'percent_increase_quantity_allowed'
+                    ]
+                    allowed_quantity *= 1 + (
+                        percent_increase_quantity_allowed / 100.
+                    )
+                if self.quantity > allowed_quantity:
```
Comment (rousseldenis) [inline stock_no_extra_move/wizard/stock_transfer_details.py:36]: @leemannd Please use also float_compare

**Review by rousseldenis — CHANGES_REQUESTED** [review]
Code review

**Comment by rousseldenis** [comment]
@leemannd

**Comment by leemannd** [comment]
Thanks for the reminder. Will take a look about it

**Comment by leemannd** [comment]
I currently have not time to have a look at it. Closing the stuff

**Pull request closed without merging**